src/providers/twitch/TwitchChannel.cpp

```diff
@@ -607,10 +607,30 @@ void TwitchChannel::setLive(bool newLiveStatus)
             }
             else
             {
+                // Channel offline message
                 MessageBuilder builder;
                 TwitchMessageBuilder::offlineSystemMessage(
                     this->getDisplayName(), &builder);
                 this->addMessage(builder.release());
+
+                // "delete" old 'CHANNEL is live' message
+                LimitedQueueSnapshot<MessagePtr> snapshot =
+                    getApp()->twitch2->liveChannel->getMessageSnapshot();
+                int snapshotLength = snapshot.size();
+
+                int end = std::max(0, snapshotLength - 200);
+                auto liveMessageSearchText =
+                    QString("%1 is live!").arg(this->getDisplayName());
+
+                for (int i = snapshotLength - 1; i >= end; --i)
```

Comment:
nit: Can these `int`s be `size_t`s? [`LimitedQueueSnapshot<T>::size()`](https://github.com/Chatterino/chatterino2/blob/5512437f1bd5b2be7997b52bf36b04930446f3f1/src/messages/LimitedQueueSnapshot.hpp#L25) returns a `size_t`

Reply:
These `int`s cannot be `size_t`s because if `snapshotLength` is <0 then it will overflow and then `std::max` will pick the obviously bigger overflowed value. Also this code was copypasted from [here](https://github.com/Chatterino/chatterino2/blob/f1e843a67275cd8e9d5a3c738a504d54b7743c42/src/common/Channel.cpp#L249)